File: 1001-1100/1021-Remove Outermost Parentheses/1021-Remove Outermost Parentheses.cpp
class Solution {
public:
    string removeOuterParentheses(string S) {
        string result;
        int bal = 0;
        for (char c : S) {
            if (c == '(') {
                if (bal++ > 0) {
                    result += c;
                }
            }
            else if (bal-- > 1) {
                result += c;
            }
        }
        return result;
    }
};

File: 1001-1100/1021-Remove Outermost Parentheses/1021-Remove Outermost Parentheses.py
class Solution:
    def removeOuterParentheses(self, S: str) -> str:
        result = []
        bal = 0
        for c in S:
            if c == '(':
                bal += 1
                if bal > 1:
                    result.append(c)
            else:
                bal -= 1
                if bal > 0:
                    result.append(c)
        return ''.join(result)
